src/Microsoft.Azure.WebJobs.Host/Executors/StorageAccountProviderExtensions.cs

```diff
@@ -54,7 +56,25 @@ public static Task<IStorageAccount> GetStorageAccountAsync(this IStorageAccountP
                 }
             }
 
-            return provider.GetAccountAsync(connectionStringName, cancellationToken);
+            IStorageAccount account = await provider.GetAccountAsync(connectionStringName, cancellationToken);
+            ValidateStorageAccount(account, connectionStringName);
+            return account;
+        }
+
+        private static void ValidateStorageAccount(IStorageAccount account, string connectionStringName)
```

Comment:
does it make sense to bump this up into GetAccountAsync, where most storage account validation occurs?